lib/bip-39.js

```diff
@@ -15,281 +15,278 @@
+  }
+
+  /**
      * Convert a mnemonic to a seed. Does not check for validity of the mnemonic -
      * for that, you should manually run check() first.
      */
-    mnemonic2Seed (passphrase = '') {
-      let mnemonic = this.mnemonic
-      if (!this.check()) {
-        throw new Error(
-          'Mnemonic does not pass the check - was the mnemonic typed incorrectly? Are there extra spaces?'
-        )
-      }
-      if (typeof passphrase !== 'string') {
-        throw new Error('passphrase must be a string or undefined')
-      }
-      mnemonic = mnemonic.normalize('NFKD')
-      passphrase = passphrase.normalize('NFKD')
-      let mbuf = Buffer.from(mnemonic)
-      let pbuf = Buffer.concat([
-        Buffer.from('mnemonic'),
-        Buffer.from(passphrase)
-      ])
-      this.seed = Kdf.pbkdf2(mbuf, pbuf, 2048, 64 * 8)
-      return this
+  mnemonic2Seed (passphrase = '') {
+    let mnemonic = this.mnemonic
+    if (!this.check()) {
+      throw new Error(
+        'Mnemonic does not pass the check - was the mnemonic typed incorrectly? Are there extra spaces?'
+      )
     }
-
-    isValid (passphrase = '') {
-      let isValid
-      try {
-        isValid = !!this.mnemonic2Seed(passphrase)
-      } catch (err) {
-        isValid = false
-      }
-      return isValid
+    if (typeof passphrase !== 'string') {
+      throw new Error('passphrase must be a string or undefined')
     }
+    mnemonic = mnemonic.normalize('NFKD')
+    passphrase = passphrase.normalize('NFKD')
+    let mbuf = Buffer.from(mnemonic)
+    let pbuf = Buffer.concat([
+      Buffer.from('mnemonic'),
+      Buffer.from(passphrase)
+    ])
+    this.seed = pbkdf2.pbkdf2Sync(mbuf, pbuf, 2048, 64, 'sha512')
+    return this
+  }
 
-    static isValid (mnemonic, passphrase = '') {
-      return new Bip39(mnemonic).isValid()
+  isValid (passphrase = '') {
+    let isValid
+    try {
+      isValid = !!this.mnemonic2Seed(passphrase)
+    } catch (err) {
+      isValid = false
     }
+    return isValid
   }
 
-  return Bip39
+  static isValid (mnemonic, passphrase = '') {
+    return new Bip39(mnemonic).isValid()
+  }
 }
 
-inject = require('./injecter')(inject, dependencies)
-let Bip39 = inject()
-Bip39.En = inject({
-  Wordlist: require('./bip-39-en')
-})
-Bip39.Jp = inject({
-  Wordlist: require('./bip-39-jp')
-})
 module.exports = Bip39
+
+/*
+  The need to export these other languages here is *questionable* at best.
+
+  This will ensure that whenever new Bip39 languages are added in the future,
+  all of them are always included in bundles that import Bip39. It may be
+  better to simply require users to import lib/bip-39-xx when they wish to use
+  a non-default language.
+
+  it would also be nice to be able to inject a default Bip39 language, similar
+  to how we inject the default NETWORK variable.
+
+  Additionally, Removal of these exports would allow as many new languages as
+  desired to be added without unduly impacting final build output size.
```

Comment:
I agree with your opinion. We should not export all languages here. It does seem like simply having a separate class for each language may be useful. That way people can use which ever one they want for which language. That would allow us to add more in a straightforward way without requiring that all users of this class get all of them.

Reply:
...however, that seems like that belongs in a separate PR.

Reply:
Actually, I didn't realize that's how it already works. There already are separate classes. Yeah, let's remove this.